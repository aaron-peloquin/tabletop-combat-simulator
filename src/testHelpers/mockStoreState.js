export default {
  "creatures": [
    {
      "name": "Goplin",
      "cr": "1/4",
      "hash": "hivn3p0k",
      "description": "A stout monster",
      "defaultInitiative": 10,
      "armor": 11,
      "hp": 5,
      "hitDiceEquation": "1d20+1",
      "damageDiceEquation": "1d4"
    },
    {
      "name": "Elven Ranger",
      "cr": "1/2",
      "hash": "h3dxf5wl",
      "description": "A swift hunter",
      "defaultInitiative": 16,
      "armor": 13,
      "hp": 10,
      "hitDiceEquation": "1d20+8",
      "damageDiceEquation": "1d6+3"
    },
    {
      "name": "Knight",
      "cr": "1",
      "hash": "hee9p286",
      "description": "An honorable soldier",
      "defaultInitiative": 11,
      "armor": 18,
      "hp": 25,
      "hitDiceEquation": "1d20+4",
      "damageDiceEquation": "1d12+4"
    },
    {
      "name": "Lizardfolk",
      "cr": "3",
      "hash": "howx26rt",
      "description": "An ancient sneaky monster",
      "defaultInitiative": 14,
      "armor": 14,
      "hp": 15,
      "hitDiceEquation": "1d20+5",
      "damageDiceEquation": "1d10+2"
    }
  ],
  "sideBar": false,
}