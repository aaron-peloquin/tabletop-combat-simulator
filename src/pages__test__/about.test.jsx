import { connect } from "react-redux"
import { mountWithStore } from "enzyme-redux"
import { createMockStore } from "redux-test-utils"
import mockStoreState from "../testHelpers/mockStoreState"
import toJson from "enzyme-to-json"

import About from "../pages/about"

describe("<About />", ()=>{
  const ReactComponent = () => <About />
  const mapStateToProps = (state) => ({state})
  const store = createMockStore(mockStoreState)
  const ConnectedComponent = connect(mapStateToProps)(ReactComponent)
  const component = mountWithStore(<ConnectedComponent />, store)

  it("loads", async () => {
    expect(typeof component).toBe("object")
  })

  it("has a NextJS prop.title of About", async () => {
    const props = await About.getInitialProps()
    expect(props.title).toBe("About")
  })

  it("Snapshots", () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})