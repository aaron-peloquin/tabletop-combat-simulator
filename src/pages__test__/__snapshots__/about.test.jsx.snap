// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Creature /> Page Snapshots 1`] = `
<Fragment>
  <WithStyles(Typography)
    variant="h2"
  >
    About
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="body2"
  >
    The goal of this website is to help simulate simple combat scenarios for your tabletop game.
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="h4"
  >
    App Code
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="body2"
  >
    The code for this web app is open-source, see 
    <a
      href="https://github.com/aaron-peloquin/tabletop-combat-simulator"
    >
      Github
    </a>
    .
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="h4"
  >
    Thanks
  </WithStyles(Typography)>
  <WithStyles(Typography)
    variant="body2"
  >
    Special thanks to 
    <a
      href="https://game-icons.net/"
      target="_blank"
    >
      Game-Icons.net
    </a>
    , who provided the icons on this app under under
    <a
      href="https://creativecommons.org/licenses/by/3.0/"
      target="_blank"
    >
      Creative Commons
    </a>
    .
  </WithStyles(Typography)>
</Fragment>
`;
